lib/variable/structures.cpp

```diff
@@ -30,6 +31,16 @@ make_structures<Eigen::Affine3d, double>(const Dimensions &,
                                          const units::Unit &,
                                          element_array<double> &&);
 
+template SCIPP_VARIABLE_EXPORT Variable
+make_structures<scipp::core::Quaternion, double>(const Dimensions &,
+                                                 const units::Unit &,
```

Comment:
It occurs to me now that we have rotation matrices with a unit in practice (UB matrices). If someone would attempt to turn this into an affine transform we have no way of handling the unit. I don't think this is a problem for now, provided that we prevent this and throw an exception. Can you ensure we have tests in place for these scenarios?